File: ros_ws/src/navigation_collisions/src/engine_shapes_attrib.py
class Velocity(object):
    def __init__(self, linear, angular):
        self.Linear = linear
        self.Angular = angular


class Point(object):
    def __init__(self, x, y):
        self.X = x
        self.Y = y

    def tuple2(self):
        return (self.X, self.Y)


class Position(Point):
    def __init__(self, x, y, angle = 0.0):
        super(Position, self).__init__(x, y)
        self.A = angle

    def tuple3(self):
        return (self.X, self.Y, self.A)

File: ros_ws/src/navigation_coppisions/src/collisions_engine/engine_shapes_attrib.cpp
#include "engine_shapes_attrib.h"


// Point class
Point::Point() {
    Point::x = 0;
    Point::y = 0;
}
Point::Point(float x, float y)
{
    Point::x = x;
    Point::y = y;
}


// Position class
Position::Position():Point()
{
    Position::a = 0;
}
Position::Position(float x, float y, float a = 0):Point(x, y)
{
    Position::a = a;
}


// Velocity class
Velocity::Velocity(float object_width, float object_height, float linear, float angular)
{
    Velocity::object_width = object_width;
    Velocity::object_height = object_height;

    Velocity::linear = linear;
    Velocity::angular = angular;
}